chasm/lib/activity/fx.go

```diff
@@ -28,4 +28,23 @@ var FrontendModule = fx.Module(
 	fx.Provide(activitypb.NewActivityServiceLayeredClient),
 	fx.Provide(NewFrontendHandler),
 	fx.Provide(resource.SearchAttributeValidatorProvider),
+	fx.Invoke(func(registry *chasm.Registry) error {
+		return registry.Register(&componentOnlyLibrary{})
+	}),
 )
+
+// componentOnlyLibrary only registers the Activity component. Used by frontend which needs to
+// serialize ComponentRefs but doesn't need task executors.
+type componentOnlyLibrary struct {
+	chasm.UnimplementedLibrary
+}
+
+func (l *componentOnlyLibrary) Name() string {
+	return "activity"
+}
+
+func (l *componentOnlyLibrary) Components() []*chasm.RegistrableComponent {
+	return []*chasm.RegistrableComponent{
+		chasm.NewRegistrableComponent[*Activity]("activity"),
+	}
+}
```

Comment:
@bergundy can you double-check this change please? The problem being addressed is that after merging `main`, the `FooByID` tests fail with:

```
unknown chasm component type: *activity.Activity
```

It seems that we moved away from a shared registry on `main`. I can try to describe this better if needed.